### DISCUSSION TITLE
Pressure Units

### DISCUSSION THREAD

**Discussion by Peter847** [jason0x43/hacs-hubitat#280] [General]
I am running Home Assistant on a HyperV instance where it talks to three Hubitat C7 hubs.  I have a pressure sensor in one of the hubs that reads in Pascals eg 158Pa but when it is displayed in Home Assistant it shows as x100 that value eg 15,800Pa.  I have changed the units in HA to Hecto Pascals (hPa) and now the numeric value is correct but, of course, the units are wrong.

Here is the start of the DeviceID using the Hubitat Maker API call:

{
  "id": "123",
  "name": "Shelly Plus 1 Test",
  "label": "0 - Furnace Room -  Furnace Power",
  "type": "Shelly Plus 1 v1",
  "room": "0 - Furnace Room",
  "attributes": [
    {
      "name": "pressure",
      "currentValue": 157.88,
      "dataType": "NUMBER"
    },

and a cut and paste from the HA Sensor Card for the device:

  Pressure 157.88 hPa

Is there a way to correct the numeric vale and/or the units?

**Comment by jason0x43** [comment]
What does the complete device information look like? Specifically, what does it report, if anything, for units?

Also, can you enable [debug logging](https://github.com/jason0x43/hacs-hubitat?tab=readme-ov-file#logging) and grab a pressure change event from the log? It will look something like this:

```
2024-08-12 15:27:49.377 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'speed', 'value': 'low', 'displayName': 'Virtual Fan', 'deviceId': '19', 'descriptionText': 'Virtual Fan fan speed set to low', 'unit': None, 'type': None, 'data': None}
```

I'm curious to see what units Hubitat is reporting for the sensor in update events.

By default the integration uses millibars for a sensor's native units, so if it's not being updated by the sensor config, HA thinks whatever value it's getting from Hubitat is millibars. When HA is told to display the sensor in Pa, it will multiply it by 100 to get Pa from millibars.

**Reply by Peter847** [reply]
Many thanks for the reply.  I am in the process of rebuilding my Hubitat network with a new hub and haven't setup the Maker API yet.  Give another week or so and I will provide the logs.

**Reply by Peter847** [reply]
Sorry it took so long to reply, I had some issues rebuilding my network on a new hub.

Here is the HA debug log for a pressure change event in the Hubitat hub:

`2024-08-24 16:57:02.328 DEBUG (MainThread) [custom_components.hubitat.device] handling event for <entity sensor.0_furnace_furnace_power_pressure=100.0> (0 - Furnace - Furnace Power Pressure, <class 'custom_components.hubitat.sensor.HubitatPressureSensor'>)`

The event is created in Hubitat through this line of Groovy code:

`sendEvent(name:"pressure",value: localPressure, unit:"Pa", desriptionText:"setPressure()")`

Hope this helps track down the issue!

**Reply by jason0x43** [reply]
Hmmmm...the send call in the groovy code certainly suggests that the units should be Pa in the event received by HA. Did you happen to see an HA event log message with the event data that was actually received by HA? It would look like

```
2024-08-12 15:27:49.377 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'speed', 'value': 'low', 'displayName': 'Virtual Fan', 'deviceId': '19', 'descriptionText': 'Virtual Fan fan speed set to low', 'unit': None, 'type': None, 'data': None}
```

I'm going to double check the sensor code; it's possible that the HA entity isn't updating its units with the units in the Hubitat event.

**Reply by jason0x43** [reply]
Oooo, I think I see where the problem is. The integration does not appear to be paying attention to units in events.

**Reply by jason0x43** [reply]
I just released v0.9.33 with support for custom pressure units -- give that a try.

**Reply by Peter847** [reply]
Apologies for the delayed response, I have been travelling.  I updated to v0.9.33 and changed the units from hPa to Pa and HA still displays the pressure as x100 of what it should be, ie 4,700Pa when the hub is sending 47Pa.  Here are the debug log lines:

2024-09-14 07:55:36.781 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'pressure', 'value': '47', 'displayName': '0 - Furnace - Furnace Power', 'deviceId': '106', 'descriptionText': None, 'unit': 'Pa', 'type': None, 'data': None}
2024-09-14 07:55:36.781 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating pressure of 106 to 47 (Pa)

**Reply by jason0x43** [reply]
Try v0.9.34 when you get a chance. It fixes a potential issue with how units are handled and also includes some additional logging statements to help diagnose what might be going wrong.

**Reply by Peter847** [reply]
That worked!  Many thanks for fixing it
